app/vmock.go

```diff
@@ -33,7 +33,10 @@ func wireValidatorMock(conf Config, pubshares []eth2p0.BLSPubKey, sched core.Sch
 		return err
 	}
 
-	onStartup := true
+	var (
+		mu        sync.Mutex
```

Comment:
please wrap accessing shared data in a function. You have a bug, you need to lock when writing AND reading.